lib/awful/widget/watch.lua

```diff
@@ -79,7 +80,7 @@ function watch.new(command, timeout, callback, base_widget)
     end)
     t:start()
     t:emit_signal("timeout")
-    return base_widget
+    return { widget = base_widget, timer = t }
```

Comment:
This changes the API and thus will be blocked by @Elv13. How about `return base_widget, t` instead? (For the API doc: You would then just have multiple `@return`-tags, in the order in which the values are returned).
If this is a good idea is for others to decide...